lib/discordrb/data/user.rb

```diff
@@ -167,6 +188,12 @@ def send_file(file, caption = nil, filename: nil, spoiler: nil)
       pm.send_file(file, caption: caption, filename: filename, spoiler: spoiler)
     end
 
+    # @return [String, nil] the ID of this user's current banner, can be used to generate a banner URL.
+    # @see #banner_url
+    def banner_id
+      @banner_id ||= JSON.parse(API::User.resolve(@bot.token, @id))['banner']
```

Comment:
This feels a bit strange. We're in the context of a user so are we guarding against the case that we have an out of date cache? If these entities are updated via the gateway we should consider not doing a separate request here